ethtx/decoders/decoder_service.py

```diff
@@ -9,19 +9,23 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-
+import logging
 from typing import Dict, List, Union
 
+from .abi.decoder import ABIDecoder
+from .semantic.decoder import SemanticDecoder
 from ..models.decoded_model import DecodedTransaction
 from ..models.objects_model import Block, Call
 
+log = logging.getLogger(__name__)
 
 class DecoderService:
-    def __init__(self, abi_decoder, semantic_decoder, web3provider, default_chain):
-        self.abi_decoder = abi_decoder
-        self.semantic_decoder = semantic_decoder
+    def __init__(self, abi_decoder: ABIDecoder, semantic_decoder: SemanticDecoder, web3provider, default_chain: str):
+        self.abi_decoder: ABIDecoder = abi_decoder
+        self.semantic_decoder: SemanticDecoder = semantic_decoder
         self.web3provider = web3provider
```

Comment:
is it possible to get `web3provider` type?

Reply:
added